tensornetwork/network_operations.py

```diff
@@ -713,3 +713,28 @@ def contract_trace_edges(node: BaseNode) -> BaseNode:
     if edge.is_trace():
       return contract_parallel(edge)
   raise ValueError('`node` has no trace edges')
+
+
+def reduced_density(traced_out_edges: Iterable[Edge]) -> Tuple[dict, dict]:
+  """
+
+  Args:
+    traced_out_edges:
+
+  Returns:
+
+  """
+
+  # Get all reachable nodes.
+  old_nodes = get_all_nodes(traced_out_edges)
```

Comment:
You'll want to also do `tn.reachable(...)` so that you get all possible edges, not just some of them.